src/Json.re

```diff
@@ -236,16 +245,39 @@ let parseString = (text, pos) => {
   (Scanf.unescaped(String.sub(text, pos, i^ - pos)), i^ + 1)
 };
 
-let parseNumber = (text, pos) => {
+let rec toCharList = (text, maxDepth) => {
+  if (String.length(text) < 2 || maxDepth < 2) {
+    [text.[0]]
+  } else {
+    [text.[0], ...toCharList(stringTail(text), maxDepth - 1)]
+  }
+};
+
+let continousDigits = (text, pos, len) => {
   let i = ref(pos);
-  let len = String.length(text);
   while (i^ < len && Char.code('0') <= Char.code(text.[i^]) && Char.code(text.[i^]) <= Char.code('9')) {
```

Comment:
I was thinking too naively with exponent numbers. I think we need to extend this to check for `e` or `E` (followed by `+`, `-` or ``), but once and only once.